modules/evm/src/runner/handler.rs

```diff
@@ -170,7 +178,27 @@ impl<'vicinity, 'config, 'meter, T: Config> Handler<'vicinity, 'config, 'meter,
 		match reason {
 			ExitReason::Succeed(s) => (s.into(), runtime.machine().return_value()),
 			ExitReason::Error(e) => (e.into(), Vec::new()),
-			ExitReason::Revert(e) => (e.into(), runtime.machine().return_value()),
+			ExitReason::Revert(e) => {
+				if let Some(err_msg) = &self.err_msg {
+					// Return precompile error message
+					// Refer to evm.rpc.lib.rs function: `decode_revert_message`
+					let mut output = vec![0u8; 36];
+					let mut message_len = [0u8; 32];
+					U256::from(err_msg.len()).to_big_endian(&mut message_len[..]);
+					output.extend_from_slice(&message_len.to_vec());
+					output.extend_from_slice(&err_msg);
+
+					log::debug!(
+						target: "evm",
+						"err_msg: {:?}, output: {:?}",
+						err_msg,
+						output,
+					);
+					(e.into(), output.to_vec())
```

Comment:
this ignores `runtime.machine().return_value()`, we need to make sure it is safe to do so. e.g. only override it if it is empty.